src/vs/workbench/services/search/common/searchExtTypes.ts

```diff
@@ -498,3 +498,25 @@ export interface FindTextInFilesOptions {
 	 */
 	afterContext?: number;
 }
+
+/*
+ * Options for following search.exclude and files.exclude settings.
+ */
+export enum ExcludeSettingOptions {
+	none = 1,
+	filesExclude = 2,
+	searchAndFilesExclude = 3,
+}
+
+/*
+ * Which locations of .gitignore and .ignore files to follow when searching.
+*/
+export enum SearchIgnoreOptions {
+	none = 1,
+	localOnly = 2,
+	localAndFollowSettings = 3,
```

Comment:
This means to follow the settings for global and parent ignore files-following. So using those settings to dictate whether to use them.

Reply:
Could the option be changed to `followIgnoreSettings` or a note be added to the docstring then, otherwise I think it's really easy to mix up these followSettings keys with the exclude enum options and/or exclude settings.

Reply:
Do you think that option 2 in this comment makes anything more clear? https://github.com/microsoft/vscode/issues/205692#issuecomment-2085804768

Reply:
Although this PR has the enum approach, I've been brainstorming better ways to go about this.